pkg/storegateway/series_refs_test.go

```diff
@@ -759,6 +759,127 @@ func TestDeduplicatingSeriesChunkRefsSetIterator_PropagatesErrors(t *testing.T)
 	assert.ErrorContains(t, chainedSet.Err(), "something went wrong")
 }
 
+func TestLimitingSeriesChunkRefsSetIterator(t *testing.T) {
+	testCases := map[string]struct {
+		sets                     []seriesChunkRefsSet
+		seriesLimit, chunksLimit int
+		upstreamErr              error
+
+		expectedSetsCount int
+		expectedErr       string
+	}{
+		"doesn't exceed limits": {
+			seriesLimit:       5,
+			chunksLimit:       5,
+			expectedSetsCount: 1,
+			sets: []seriesChunkRefsSet{
+				{series: []seriesChunkRefs{
+					{lset: labels.FromStrings("l1", "v1"), chunks: make([]seriesChunkRef, 1)},
+					{lset: labels.FromStrings("l1", "v2"), chunks: make([]seriesChunkRef, 1)},
+					{lset: labels.FromStrings("l2", "v1"), chunks: make([]seriesChunkRef, 1)},
+					{lset: labels.FromStrings("l2", "v2"), chunks: make([]seriesChunkRef, 1)},
+				}},
+			},
+		},
+		"exceeds chunks limit": {
+			seriesLimit:       5,
+			chunksLimit:       3,
+			expectedSetsCount: 0,
+			expectedErr:       "exceeded chunks limit",
+			sets: []seriesChunkRefsSet{
+				{series: []seriesChunkRefs{
+					{lset: labels.FromStrings("l1", "v1"), chunks: make([]seriesChunkRef, 1)},
```

Comment:
In these unit tests, all series have only 1 chunk. There's no actual difference between series limit and chunks limit. If you have a bug in the implementation and the chunks limit is implemented as series limit, then these tests won't spot it. I would diversify tests, and have series a with different number of chunks.